lib/matplotlib/widgets.py

```diff
@@ -485,21 +514,290 @@ def on_changed(self, func):
+                verticalalignment="center",
+                horizontalalignment="right",
+            )
+
+            fmted = self._format(valinit)
+            s = rf"({fmted[0]}, {fmted[1]})"
+            self.valtext = ax.text(
+                1.02,
+                0.5,
+                s,
+                transform=ax.transAxes,
+                verticalalignment="center",
+                horizontalalignment="left",
+            )
+
+        self._observers = cbook.CallbackRegistry()
+
+        self.set_val(valinit)
+
+    def _min_in_bounds(self, min):
+        """
+        Ensure the new min value is between valmin and self.val[1]
+        """
+        if min <= self.valmin:
+            if not self.closedmin:
+                return
+            min = self.valmin
+
+        if min >= self.val[1]:
+            min = self.val[1]
+        return self._stepped_value(min)
+
+    def _max_in_bounds(self, max):
+        """
+        Ensure the new max value is between valmax and self.val[0]
+        """
+        if max >= self.valmax and self.closedmax:
+            max = self.valmax
+
+        if max <= self.val[0]:
+            max = self.val[0]
+        return self._stepped_value(max)
+
+    def _value_in_bounds(self, val):
+        return (self._min_in_bounds(val[0]), self._max_in_bounds(val[1]))
+
+    def _update_val_from_pos(self, pos):
+        """
+        Given a position update the *val*
+        """
+        idx = np.argmin(np.abs(self.val - pos))
+        if idx == 0:
+            val = self._min_in_bounds(pos)
+            self.set_min(val)
+        else:
+            val = self._max_in_bounds(pos)
+            self.set_max(val)
+
+    def _update(self, event):
+        """Update the slider position."""
+        if self.ignore(event) or event.button != 1:
+            return
+
+        if event.name == "button_press_event" and event.inaxes == self.ax:
+            self.drag_active = True
+            event.canvas.grab_mouse(self.ax)
+
+        if not self.drag_active:
+            return
+
+        elif (event.name == "button_release_event") or (
+            event.name == "button_press_event" and event.inaxes != self.ax
+        ):
+            self.drag_active = False
+            event.canvas.release_mouse(self.ax)
+            return
+        if self.orientation == "vertical":
+            val = self._update_val_from_pos(event.ydata)
+        else:
+            val = self._update_val_from_pos(event.xdata)
+        if val not in [None, self.val]:
+            self.set_val(val)
+
+    def _format(self, val):
+        """Pretty-print *val*."""
+        if self.valfmt is not None:
+            return (self.valfmt % val[0], self.valfmt % val[1])
+        else:
+            s1 = super()._format(val[0])
+            s2 = super()._format(val[1])
+            # fmt.get_offset is actually the multiplicative factor, if any.
+            return (s1 + self._fmt.get_offset(), s2 + self._fmt.get_offset())
```

Comment:
I think it is saved by the `slidermin` and `slidermax`. It would be nice to pass both values simultaneously to `format_ticks` so I'll just make `_format` not part of the superclass. There has to be same branching on `self.valfmt` in RangeSlider anyway so theres not much loss in each slider implementing `_format`.